met_connectors/src/frost/spatial.rs

```diff
@@ -97,22 +98,30 @@ fn parse_polygon(polygon: &Polygon) -> String {
 fn json_to_spatial_cache(
     resp: serde_json::Value,
     timestamp: DateTime<Utc>,
-) -> Result<SpatialCache, Error> {
+) -> Result<DataCache, Error> {
     let data = extract_data(resp, timestamp)?;
 
     let lats: Vec<f32> = data.iter().map(|d| d.1.latitude).collect();
     let lons: Vec<f32> = data.iter().map(|d| d.1.longitude).collect();
     let elevs: Vec<f32> = data.iter().map(|d| d.1.elevation).collect();
-    let values: Vec<f32> = data.iter().map(|d| d.0.body.value).collect();
+    let values: Vec<Option<f32>> = data.iter().map(|d| Some(d.0.body.value)).collect();
 
-    Ok(SpatialCache::new(lats, lons, elevs, values))
+    Ok(DataCache::new(
+        lats,
+        lons,
+        elevs,
+        Timestamp(0),
+        RelativeDuration::minutes(5),
```

Comment:
Again, you don't need to make these up, you can parse them from the response like we do in frost/series.rs. 

The only complication here is that each data point here has its own time_resolution. Ultimately we want to tweak the request (or filter the response if that isn't possible) to only get data with the same time_resolution. For now I think its fine if you just use the time resolution from the first data point, just leave a todo so we remember to fix it later (probably when we unify get_series and get_spatial)

Reply:
Also, if you're touching the duration parsing...

This code uses a custom iso8601 duration parser I wrote locally in this crate (frost/duration.rs), but I recently got a more comprehensive parser/formatter [merged upstream in chronoutil](https://github.com/olliemath/chronoutil/pull/14). So if you're up for it, you can delete that code, and replace its usage with the upstream parser